src/librustc/hir/lowering.rs

```diff
@@ -862,9 +865,14 @@ impl<'a> LoweringContext<'a> {
                 PatKind::Wild => hir::PatKind::Wild,
                 PatKind::Ident(ref binding_mode, pth1, ref sub) => {
                     self.with_parent_def(p.id, |this| {
+                        let name = match this.resolver.get_resolution(p.id).map(|d| d.full_def()) {
+                            // Only pattern bindings are renamed
```

Comment:
We don't want to rename constants or unit variants/structs, that can also be represented with `PatKind::Ident` due to parsing ambiguities, only fresh bindings.